src/azure-cli/azure/cli/command_modules/acs/_roleassignments.py

```diff
@@ -0,0 +1,168 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import time
+import uuid
+
+from azure.cli.core.azclierror import AzCLIError
+from azure.cli.core.profiles import ResourceType, get_sdk
+from azure.core.exceptions import HttpResponseError
+from azure.graphrbac.models import GetObjectsParameters
+from knack.log import get_logger
+from msrestazure.azure_exceptions import CloudError
+
+from ._client_factory import (get_auth_management_client,
+                              get_graph_rbac_management_client)
+
+logger = get_logger(__name__)
+
+
+def _get_object_stubs(graph_client, assignees):
+    params = GetObjectsParameters(include_directory_object_references=True,
+                                  object_ids=assignees)
+    return list(graph_client.objects.get_objects_by_object_ids(params))
+
+
+def resolve_object_id(cli_ctx, assignee):
+    client = get_graph_rbac_management_client(cli_ctx)
+    result = None
+    if assignee is None:
+        raise AzCLIError('Inputted parameter "assignee" is None.')
+    if assignee.find('@') >= 0:  # looks like a user principal name
+        result = list(client.users.list(
+            filter="userPrincipalName eq '{}'".format(assignee)))
+    if not result:
+        result = list(client.service_principals.list(
+            filter="servicePrincipalNames/any(c:c eq '{}')".format(assignee)))
+    if not result:  # assume an object id, let us verify it
+        result = _get_object_stubs(client, [assignee])
+
+    # 2+ matches should never happen, so we only check 'no match' here
+    if not result:
+        raise AzCLIError(
+            "No matches in graph database for '{}'".format(assignee))
+
+    return result[0].object_id
+
+
+def resolve_role_id(role, scope, definitions_client):
```

Comment:
For reference, [this](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/custom.py#L1512) is the corresponding implementation in `custom.py`.